include/aws/event-stream/event_stream_rpc_client.h

```diff
@@ -0,0 +1,181 @@
+#ifndef AWS_EVENT_STREAM_RPC_CLIENT_H
+#define AWS_EVENT_STREAM_RPC_CLIENT_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/event-stream/event_stream_rpc.h>
+
+struct aws_channel;
+struct aws_event_stream_rpc_client_connection;
+struct aws_event_stream_rpc_client_continuation_token;
+
+/**
+ * Invoked when a connection receives a message on an existing stream. message_args contains the
+ * message data.
+ */
+typedef void(aws_event_stream_rpc_client_stream_continuation_fn)(
+    struct aws_event_stream_rpc_client_continuation_token *token,
+    const struct aws_event_stream_rpc_message_args *message_args,
+    void *user_data);
+
+/**
+ * Invoked when a continuation has either been closed with the TERMINATE_STREAM flag, or when the connection
+ * shuts down and deletes the continuation.
+ */
+typedef void(aws_event_stream_rpc_client_stream_continuation_closed_fn)(
+    struct aws_event_stream_rpc_client_continuation_token *token,
+    void *user_data);
+
+struct aws_event_stream_rpc_client_stream_continuation_options {
+    aws_event_stream_rpc_client_stream_continuation_fn *on_continuation;
+    aws_event_stream_rpc_client_stream_continuation_closed_fn *on_continuation_closed;
+    void *user_data;
+};
+
+/**
+ * Invoked when a non-stream level message is received on a connection.
+ */
+typedef void(aws_event_stream_rpc_client_connection_protocol_message_fn)(
+    struct aws_event_stream_rpc_client_connection *connection,
+    const struct aws_event_stream_rpc_message_args *message_args,
+    void *user_data);
+
+/**
+ * Invoked when a successfully created connection is shutdown. error_code will indicate the reason for the shutdown.
+ */
+typedef void(aws_event_stream_rpc_client_on_connection_shutdown_fn)(
+    struct aws_event_stream_rpc_client_connection *connection,
+    int error_code,
+    void *user_data);
+
+typedef int(aws_event_stream_rpc_client_on_connection_setup_fn)(
```

Comment:
Comment whether user needs to acquire() reference on connection or not.

Also, comment what happens if user returns non-zero. Should they ALSO release their reference to the connection? Or does returning non-zero indicate that they never took responsibility for it? Will the shutdown still fire?

comment what happens if user returns non-zero.
Does the user still need to release() the connection? Or

Reply:
actually, you're never checking this return value
did you actually intend to have a return value for this function?

Reply:
no if it has a return value I'm supposed to be using it, it's a bug.

Reply:
woops, wrong function. Yes, this one can be void.